src/graph.rs

```diff
@@ -3426,19 +3426,38 @@ impl<'a, 'graph> Builder<'a, 'graph> {
     package_req: &PackageReq,
   ) -> Option<Version> {
     // try to find in the list of existing versions first
-    if let Some(existing_versions) =
+    let versions = if let Some(existing_versions) =
       self.graph.packages.versions_by_name(&package_req.name)
     {
-      if let Some(version) = resolve_version(
-        &package_req.version_req,
-        existing_versions.iter().map(|nv| &nv.version),
-      ) {
-        return Some(version.clone());
-      }
+      existing_versions
+        .iter()
+        .map(|nv| &nv.version)
+        .collect::<Vec<_>>()
+    } else {
+      package_info.versions.keys().collect::<Vec<_>>()
+    };
+    let unyanked_versions = versions.clone().into_iter().filter(|v| {
```

Comment:
Is cloning the versions necessary here? I'm guessing necessary because of the borrow checker somehow.

Reply:
It's just a vec of refs, yeah it's to avoid a double ref on the versions in the iterator. Could be done by mapping as well.

Reply:
Changed to `.iter().copied()`.